src/reactviews/pages/PublishProject/index.tsx

```diff
@@ -0,0 +1,15 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import ReactDOM from "react-dom/client";
+import "../../index.css";
+import { VscodeWebviewProvider } from "../../common/vscodeWebviewProvider";
+import PublishProjectPage from "./publishProject";
+
+ReactDOM.createRoot(document.getElementById("root")!).render(
+    <VscodeWebviewProvider>

```

Comment:
use VscodeWebviewProvider2 and the selector mechanism to access state. This helps in keeping the renders local to the component and improve perf. 

To see examples you can see the execution plan (simpler) and query results (a little more complex). 

Here is the index file https://github.com/microsoft/vscode-mssql/blob/main/src/reactviews/pages/ExecutionPlan/index.tsx

Here is the state provider: https://github.com/microsoft/vscode-mssql/blob/main/src/reactviews/pages/ExecutionPlan/executionPlanStateProvider.tsx

and here is how to access a state prop: https://github.com/microsoft/vscode-mssql/blob/425fb9a77db99eee37e2cb65ed2156b8b5e9a7df/src/reactviews/pages/QueryResult/queryResultPane.tsx#L174-L175

Reply:
oh! didn't notice there is a latest one. will definitely look into the latest webView provider and make the required updates. Thanks